[18.0][MIG] pos_payment_terminal: Migration to 18.0

P-H-Phuc opened the pull request:
**UPDATED**
- Refactor js file
- Remove pos_session.py: 
    -  `_loader_params_pos_payment_method` is deprecated, use `_load_pos_data_fields` in `pos.payment.method` model.
- Using `register_payment_method` instead of.

**NEW FEATURES**
- Add "OCA Payment Terminal" field to use the Proxy.
- Add "Hide Back Button" field to prevent from returning to basket from payment screen when a  payment line exists.
- Add "Fast Payment" field to auto send request to the terminal.

**CONFIGUATION**
- Go to Point of Sale > Configuration > Payment Methods
<img width="820" height="323" alt="image" src="https://github.com/user-attachments/assets/096be271-cc7e-45df-89cc-010e25b04e66" />

- Go to Point of Sale > Configuration > Settings, add payment method to PoS and config IoT Box
<img width="1228" height="347" alt="2025-12-22_17-25" src="https://github.com/user-attachments/assets/23aa3d53-bb27-4f37-bdbc-00f5aa902305" />
<img width="681" height="468" alt="image" src="https://github.com/user-attachments/assets/03b6aef1-a4aa-468f-9451-9c9ad49b695c" />

yvaucher committed:
Update module info

Add OCA as author of OCA addons

In order to get visibility on https://www.odoo.com/apps the OCA board has
decided to add the OCA as author of all the addons maintained as part of the
association.

pos_payment_terminal: Add POT file, FR translation and demo data

Better XML header

Commit:
pos_customer_display: Add button "Display Total to Customer" and port to new API
pos_payment_terminal: code cleanup

alexis-via committed:
pos_payment_terminal: smiplify JS code

Move description from __openerp__.py to README.rst
Update demo data

FIX my previous commit: sent currency to POSbox

[FIX] bug #36 : take advantage of the recent update in the odoo codebase to properly get the 'name' of the currency

sbidoul committed:
[UPD] prefix versions with 8.0

[MIG] Make modules uninstallable

[MIG] Rename manifest files

yuntux committed:
begin of porting pos_payment_terminal to Odoo 9.0

yuntux committed:
overload render_paymentlines

yuntux committed:
FIX : multiple calls to action on 'start transaction' click event

yuntux committed:
Add static/src/xml/templates.xml

alexis-via committed:
FIX crash on demo data

alexis-via committed:
Port pos_payment_terminal to v10

OCA Transbot updated translations from Transifex

OCA Transbot updated translations from Transifex

OCA Transbot updated translations from Transifex

OCA Transbot updated translations from Transifex

[FIX] fixes currencies with decimals <>2

OCA Transbot updated translations from Transifex

remove obsolete .pot files [ci skip]

[UPD] Update pos_payment_terminal.pot

legalsylvain committed:
new module pos_margin

etobella committed:
[ADD] PoS session pay invoice module

JordiBForgeFlow committed:
Migration of pos_session_pay_invoice to 11.0

andreparames committed:
pos_payment_terminal: receive transaction refs

Provide a mechanism to send the order UID to the payment terminal,
which can then pass the transaction reference generated by the
payment provider back to Odoo, and which is then added to the
order payment lines. This allows their subsequent reconciliation.
The order is also automatically validated when the payment
finishes.

andreparames committed:
pos_payment_terminal: inform user of transaction status

andreparames committed:
pos_payment_terminal: use currency precision

andreparames committed:
pos_payment_terminal: auto-validate orders with mixed payments

[FIX] pos_payment_terminal: Fix tests

[UPD] Update pos_payment_terminal.pot

Update translation files

Updated by "Update PO files to match POT (msgmerge)" hook in Weblate.

Translation: pos-10.0/pos-10.0-pos_payment_terminal
Translate-URL: https://translation.odoo-community.org/projects/pos-10-0/pos-10-0-pos_payment_terminal/

PierrickBrun committed:
[ADD] backport session summary report with fr translation

bishalgit committed:
Fix account-payment's github repo url

hparfr committed:
Fix JS loading : module should only be loaded in pos
Fix indentation

Increment version number

[ADD] icon.png

Commit:
[MIG] pos_margin:Migration to 11.0

Commit:
[IMP] Add pos margin in pos order report

Commit:
[12.0][MIG] pos_session_pay_invoice

legalsylvain committed:
[MIG] pos_margin: Migration to 12.0

meodoo committed:
[MIG] pos_margin: Migration to 13.0

wpichler committed:
[MIG] Migrated to Odoo v 12

pedrocs-exo committed:
Added translation using Weblate (Portuguese)

mtbochaca committed:
Added translation using Weblate (Spanish)

mtbochaca committed:
Added translation using Weblate (Catalan)

robyf70 committed:
[NEW][10.0] POS order remove line

robyf70 committed:
[MIG][12.0] POS order remove line

CarlesAntoli committed:
Added translation using Weblate (Catalan)

legalsylvain committed:
[MIG] pos_payment_terminal

legalsylvain committed:
[MIG][12.0] pos_payment_terminal

Update pos_payment_terminal/static/src/js/devices.js

Co-Authored-By: Iván Todorovich <ivan.todorovich@druidoo.io>

[REF] rename field

[UPD] Update pos_payment_terminal.pot

[UPD] README.rst

Update translation files

Updated by "Update PO files to match POT (msgmerge)" hook in Weblate.

Translation: pos-12.0/pos-12.0-pos_payment_terminal
Translate-URL: https://translation.odoo-community.org/projects/pos-12-0/pos-12-0-pos_payment_terminal/

[UPD] Update pos_payment_terminal.pot

Update translation files

Updated by "Update PO files to match POT (msgmerge)" hook in Weblate.

Translation: pos-12.0/pos-12.0-pos_payment_terminal
Translate-URL: https://translation.odoo-community.org/projects/pos-12-0/pos-12-0-pos_payment_terminal/

Translated using Weblate (Spanish)

Currently translated at 100.0% (10 of 10 strings)

Translation: pos-12.0/pos-12.0-pos_payment_terminal
Translate-URL: https://translation.odoo-community.org/projects/pos-12-0/pos-12-0-pos_payment_terminal/es/

Update translation files

Updated by "Update PO files to match POT (msgmerge)" hook in Weblate.

Translation: pos-12.0/pos-12.0-pos_payment_terminal
Translate-URL: https://translation.odoo-community.org/projects/pos-12-0/pos-12-0-pos_payment_terminal/

joanqubiq committed:
[IMP] pos_order_remove_line: black, isort, prettier

legalsylvain committed:
[IMP] pos_payment_terminal : improve display of the loading button

[UPD] Update pos_payment_terminal.pot

danimv5 committed:
Added translation using Weblate (Spanish)

OCA-git-bot committed:
pos_payment_terminal 12.0.0.1.2

Update translation files

Updated by "Update PO files to match POT (msgmerge)" hook in Weblate.

Translation: pos-12.0/pos-12.0-pos_payment_terminal
Translate-URL: https://translation.odoo-community.org/projects/pos-12-0/pos-12-0-pos_payment_terminal/

[UPD] Update pos_payment_terminal.pot

Update translation files

Updated by "Update PO files to match POT (msgmerge)" hook in Weblate.

Translation: pos-12.0/pos-12.0-pos_payment_terminal
Translate-URL: https://translation.odoo-community.org/projects/pos-12-0/pos-12-0-pos_payment_terminal/

legalsylvain committed:
[IMP] pos_payment_terminal : improve the display of the button start transaction

[UPD] Update pos_payment_terminal.pot

OCA-git-bot committed:
pos_payment_terminal 12.0.0.1.3

oca-transbot committed:
Update translation files

Updated by "Update PO files to match POT (msgmerge)" hook in Weblate.

Translation: pos-12.0/pos-12.0-pos_payment_terminal
Translate-URL: https://translation.odoo-community.org/projects/pos-12-0/pos-12-0-pos_payment_terminal/

[UPD] Update pos_payment_terminal.pot

Update translation files

Updated by "Update PO files to match POT (msgmerge)" hook in Weblate.

Translation: pos-12.0/pos-12.0-pos_payment_terminal
Translate-URL: https://translation.odoo-community.org/projects/pos-12-0/pos-12-0-pos_payment_terminal/

jguenat committed:
Added translation using Weblate (French)

alexis-via committed:
[MIG] pos_payment_terminal to v14

It is a full re-write of the module, in order to take advantage of the new
payment interface JS framework of Odoo POS
Update FR translation

alexis-via committed:
pos_payment_terminal: black, isort and pre-commit stuff

pos_payment_terminal: update README and screenshots

pos_payment_terminal: remove commented lines

pos_payment_terminal: fix lint issues

acsonefho committed:
[IMP] pos_session_pay_invoice: black, isort, prettier

acsonefho committed:
[13.0][MIG] pos_session_pay_invoice

sbidoul committed:
pos_payment_terminal: api conformity

Have send_payment_request return a promise as requested by the
documentation. This open the door to handling the transaction status
reported by the terminal driver asynchronously.

Also, catch errors when sending the transaction start command.

sbidoul committed:
pos_payment_terminal: reject transactions with zero amount

sbidoul committed:
pos_payment_terminal: remove dead code

[UPD] Update pos_payment_terminal.pot

[UPD] README.rst

sbidoul committed:
[IMP] pos_payment_terminal: handle terminal transaction statuses

sbidoul committed:
[FIX] pos_payment_terminal: fix error handling

pos.proxy.message does not raise a proper error apparently,
so we can't access error.stack.

sbidoul committed:
[IMP] pos_payment_terminal: support multiple terminals

sbidoul committed:
pos_payment_terminal: update authors

sbidoul committed:
[FIX] pos_payment_terminal: activate proxy if oca payment terminal is used

Without this, the driver proxy would only activate if another device
is enabled in the POS config (such as a printer or customer display).
This prevented using a configuration where the oca payment terminal
is the only connected device.

sbidoul committed:
[FIX] pos_payment_terminal: typo in js test

[UPD] Update pos_payment_terminal.pot

[UPD] README.rst

OCA-git-bot committed:
pos_payment_terminal 14.0.2.0.0

Update translation files

Updated by "Update PO files to match POT (msgmerge)" hook in Weblate.

Translation: pos-14.0/pos-14.0-pos_payment_terminal
Translate-URL: https://translation.odoo-community.org/projects/pos-14-0/pos-14-0-pos_payment_terminal/

ivantodorovich committed:
[FIX] Use scss stylesheet instead of less

chienandalu committed:
[ADD] pos_payment_method_cashdro: New module

This module allows to make payments with a Cashdro (https://www.cashdro.com) terminal
on the Point of Sale frontend.

TT29224

dsolanki-initos committed:
[IMP] pos_margin: isort, black, prettier.

dsolanki-initos committed:
[MIG] Migrate module pos_margin to v14.

legalsylvain committed:
[IMP] pos_margin : display margin and margin rate in front-end UI

Co-authored-by: Iván Todorovich <ivan.todorovich@gmail.com>

olgamarcocb committed:
[MIG] pos_session_pay_invoice: Ending migration to 13.0

fshah-initos committed:
[IMP] pos_order_remove_line: isort, black , prettier.

fshah-initos committed:
[MIG] Migrate module pos_order_remove_line to v14.

sbidoul committed:
pos_margin: bump version to workround PyPI glitch

Commit:
[IMP] pos_report_session_summary: black, isort, prettier

Commit:
[IMP] pos_payment_terminal: Graceful cancel handling

Co-authored-by: Tom Blauwendraat <tom@sunflowerweb.nl>

[FIX] pos_payment_terminal: reject cancellation

Since we can't cancel the transaction, we return a rejected
promise. The user should cancel manually which will
effectively free the transaction and let the user delete it.

[UPD] Update pos_payment_terminal.pot

Commit:
[MIG] pos_report_session_summary: Migration to 14.0

Commit:
[FIX] pos_report_session_summary

olgamarcocb committed:
[IMP] pos_session_pay_invoice: black, isort, prettier

Commit:
[FIX] pos_report_session_summary: do not crash if statement data is missing

olgamarcocb committed:
[MIG] pos_session_pay_invoice: Migration to 14.0

oca-travis committed:
[UPD] Update pos_session_pay_invoice.pot

francesco-ooops committed:
Translated using Weblate (Italian)

Currently translated at 37.5% (3 of 8 strings)

Translation: pos-14.0/pos-14.0-pos_margin
Translate-URL: https://translation.odoo-community.org/projects/pos-14-0/pos-14-0-pos_margin/it/

OCA-git-bot committed:
pos_payment_terminal 14.0.2.1.0

Update translation files

Updated by "Update PO files to match POT (msgmerge)" hook in Weblate.

Translation: pos-14.0/pos-14.0-pos_payment_terminal
Translate-URL: https://translation.odoo-community.org/projects/pos-14-0/pos-14-0-pos_payment_terminal/

Translated using Weblate (Italian)

Currently translated at 21.4% (3 of 14 strings)

Translation: pos-14.0/pos-14.0-pos_payment_terminal
Translate-URL: https://translation.odoo-community.org/projects/pos-14-0/pos-14-0-pos_payment_terminal/it/

[IMP] pos_payment_terminal: black, isort, prettier

cvinh committed:
[FIX] Hide margin in Front End if config is false

chienandalu committed:
[IMP] pos_payment_method_cashdro: black, isort, prettier

chienandalu committed:
[MIG] pos_payment_method_cashdro: Migration to 14.0

TT38070

chienandalu committed:
[FIX] pos_payment_method_cashdro: floating point quirks

Overcome floting point issues: https://stackoverflow.com/a/588014

For expample 2.3 * 100 ->  229.99999999999997

TT42574

nwinkelstraeter committed:
[MIG] pos_payment_terminal: Migration to 16.0

- Moved asset definiton to manifest
- pos.proxy was moved to pos.env.proxy
- Adapted models.js to new class structure of point_of_sale.models

[UPD] Update pos_payment_terminal.pot

[UPD] README.rst

pos_payment_terminal 16.0.1.0.1

Translated using Weblate (Italian)

Currently translated at 100.0% (14 of 14 strings)

Translation: pos-16.0/pos-16.0-pos_payment_terminal
Translate-URL: https://translation.odoo-community.org/projects/pos-16-0/pos-16-0-pos_payment_terminal/it/

Translated using Weblate (Italian)

Currently translated at 100.0% (14 of 14 strings)

Translation: pos-16.0/pos-16.0-pos_payment_terminal
Translate-URL: https://translation.odoo-community.org/projects/pos-16-0/pos-16-0-pos_payment_terminal/it/

Translated using Weblate (Spanish)

Currently translated at 28.5% (4 of 14 strings)

Translation: pos-16.0/pos-16.0-pos_payment_terminal
Translate-URL: https://translation.odoo-community.org/projects/pos-16-0/pos-16-0-pos_payment_terminal/es/

Translated using Weblate (Italian)

Currently translated at 100.0% (14 of 14 strings)

Translation: pos-16.0/pos-16.0-pos_payment_terminal
Translate-URL: https://translation.odoo-community.org/projects/pos-16-0/pos-16-0-pos_payment_terminal/it/

Translated using Weblate (Spanish)

Currently translated at 100.0% (14 of 14 strings)

Translation: pos-16.0/pos-16.0-pos_payment_terminal
Translate-URL: https://translation.odoo-community.org/projects/pos-16-0/pos-16-0-pos_payment_terminal/es/

[UPD] README.rst

Update translation files

Updated by "Update PO files to match POT (msgmerge)" hook in Weblate.

Translation: pos-16.0/pos-16.0-pos_payment_terminal
Translate-URL: https://translation.odoo-community.org/projects/pos-16-0/pos-16-0-pos_payment_terminal/

Translated using Weblate (Spanish)

Currently translated at 100.0% (12 of 12 strings)

Translation: pos-16.0/pos-16.0-pos_payment_terminal
Translate-URL: https://translation.odoo-community.org/projects/pos-16-0/pos-16-0-pos_payment_terminal/es/

Translated using Weblate (Italian)

Currently translated at 100.0% (12 of 12 strings)

Translation: pos-16.0/pos-16.0-pos_payment_terminal
Translate-URL: https://translation.odoo-community.org/projects/pos-16-0/pos-16-0-pos_payment_terminal/it/

[FIX] eslint error 'no-shadow' in all modules

[FIX] eslint error 'eqeqeq' in all modules

[BOT] post-merge updates

[FIX] check current config payment methods to force or not the use of the proxy + raise an error when its not properly configured

[UPD] Update pos_payment_terminal.pot

[BOT] post-merge updates

Update translation files

Updated by "Update PO files to match POT (msgmerge)" hook in Weblate.

Translation: pos-16.0/pos-16.0-pos_payment_terminal
Translate-URL: https://translation.odoo-community.org/projects/pos-16-0/pos-16-0-pos_payment_terminal/

Translated using Weblate (Italian)

Currently translated at 100.0% (13 of 13 strings)

Translation: pos-16.0/pos-16.0-pos_payment_terminal
Translate-URL: https://translation.odoo-community.org/projects/pos-16-0/pos-16-0-pos_payment_terminal/it/

francesco-ooops committed:
Translated using Weblate (Italian)

Currently translated at 37.5% (3 of 8 strings)

Translation: pos-14.0/pos-14.0-pos_margin
Translate-URL: https://translation.odoo-community.org/projects/pos-14-0/pos-14-0-pos_margin/it/

suker committed:
[IMP] : pre-commit stuff

suker committed:
[MIG] pos_margin: Migration to 16.0

suker committed:
[MIG] pos_order_remove_line: Migration to 16.0

mymage committed:
Added translation using Weblate (Italian)

mymage committed:
Added translation using Weblate (Italian)

mymage committed:
Added translation using Weblate (Italian)

Ricardoalso committed:
[MIG] pos_report_session_summary: Migration to 15.0

mymage committed:
Added translation using Weblate (Italian)

Commit:
[14.0][IMP] pos_session_pay_invoice: Set `target=new` on views

Commit:
[FIX] pos_session_pay_invoice: `search_paid_order_ids` must exclude orders createds with this module

Commit:
[FIX] `search_paid_order_ids` must exclude orders createds with this module

Commit:
[UPD] Update pos_session_pay_invoice.pot

mymage committed:
Translated using Weblate (Italian)

Currently translated at 100.0% (30 of 30 strings)

Translation: pos-14.0/pos-14.0-pos_session_pay_invoice
Translate-URL: https://translation.odoo-community.org/projects/pos-14-0/pos-14-0-pos_session_pay_invoice/it/

mymage committed:
Translated using Weblate (Italian)

Currently translated at 100.0% (8 of 8 strings)

Translation: pos-14.0/pos-14.0-pos_payment_method_cashdro
Translate-URL: https://translation.odoo-community.org/projects/pos-14-0/pos-14-0-pos_payment_method_cashdro/it/

OCA-git-bot committed:
pos_session_pay_invoice 14.0.1.1.0

suker committed:
[FIX] pos_order_remove_line: clean refunded orderline of toRefundLines when deleting an orderline from remove-line-button.

mymage committed:
Added translation using Weblate (Italian)

mymage committed:
Translated using Weblate (Italian)

Currently translated at 100.0% (8 of 8 strings)

Translation: pos-16.0/pos-16.0-pos_margin
Translate-URL: https://translation.odoo-community.org/projects/pos-16-0/pos-16-0-pos_margin/it/

anmarmo1 committed:
Translated using Weblate (Spanish)

Currently translated at 86.6% (26 of 30 strings)

Translation: pos-14.0/pos-14.0-pos_session_pay_invoice
Translate-URL: https://translation.odoo-community.org/projects/pos-14-0/pos-14-0-pos_session_pay_invoice/es/

Commit:
[16.0][FIX] pos_order_remove_line: Trigger

- Function logic is modified in order to use the numpad backspace trigger.
- Stops event propagation and prevents default actions.

weblate committed:
Update translation files

Updated by "Update PO files to match POT (msgmerge)" hook in Weblate.

Translation: pos-15.0/pos-15.0-pos_report_session_summary
Translate-URL: https://translation.odoo-community.org/projects/pos-15-0/pos-15-0-pos_report_session_summary/

chienandalu committed:
[FIX] *: Linter fixes

almumu committed:
Is not pIs not possible to print the report. This pr fixes the error: 'str' object has no attribute 'decimal_places' when printing session summaryossible to print the report. This pr fixes the error: 'str' object has no attribute 'decimal_places' when printing session summary

OCA-git-bot committed:
[BOT] post-merge updates

legalsylvain committed:
[FIX] eslint error 'no-shadow' in all modules

legalsylvain committed:
[FIX] eslint error 'no-shadow' in all modules

chienandalu committed:
[MIG] pos_payment_method_cashdro: migration to 16.0

TT45180

mymage committed:
Translated using Weblate (Italian)

Currently translated at 100.0% (6 of 6 strings)

Translation: pos-16.0/pos-16.0-pos_payment_method_cashdro
Translate-URL: https://translation.odoo-community.org/projects/pos-16-0/pos-16-0-pos_payment_method_cashdro/it/

Ivorra78 committed:
Translated using Weblate (Spanish)

Currently translated at 100.0% (30 of 30 strings)

Translation: pos-14.0/pos-14.0-pos_session_pay_invoice
Translate-URL: https://translation.odoo-community.org/projects/pos-14-0/pos-14-0-pos_session_pay_invoice/es/

sbejaoui committed:
[17.0][MIG] pos_payment_method_cashdro

carlos-lopez-tecnativa committed:
[MIG] pos_session_pay_invoice: Migration to 16.0

Commit:
[IMP] pos_report_session_summary: black, isort, prettier

Commit:
[MIG] pos_report_session_summary: Migration to 16.0

Commit:
[UPD] Update pos_report_session_summary.pot

OCA-git-bot committed:
[BOT] post-merge updates

Commit:
[UPD] Update pos_session_pay_invoice.pot

OCA-git-bot committed:
[BOT] post-merge updates

BhaveshHeliconia committed:
[IMP] pos_order_remove_line: pre-commit auto fixes

BhaveshHeliconia committed:
[MIG] pos_order_remove_line: Migration to 18.0

sbejaoui committed:
[IMP] pos_payment_method_cashdro: don't force https

Give the user the ability to chose between http and https.

sbejaoui committed:
[IMP] pos_payment_method_cashdro: manage refund

antoniodavid committed:
[IMP] pos_margin: pre-commit execution

antoniodavid committed:
[MIG] pos_margin: Migration to 17.0

P-H-Phuc committed:
[IMP] pos_report_session_summary: pre-commit auto fixes

P-H-Phuc committed:
[MIG] pos_report_session_summary: Migration to 18.0

mymage committed:
Translated using Weblate (Italian)

Currently translated at 100.0% (30 of 30 strings)

Translation: pos-16.0/pos-16.0-pos_session_pay_invoice
Translate-URL: https://translation.odoo-community.org/projects/pos-16-0/pos-16-0-pos_session_pay_invoice/it/

Commit:
[UPD] Update pos_session_pay_invoice.pot

OCA-git-bot committed:
[BOT] post-merge updates

weblate committed:
Update translation files

Updated by "Update PO files to match POT (msgmerge)" hook in Weblate.

Translation: pos-16.0/pos-16.0-pos_session_pay_invoice
Translate-URL: https://translation.odoo-community.org/projects/pos-16-0/pos-16-0-pos_session_pay_invoice/

mymage committed:
Translated using Weblate (Italian)

Currently translated at 100.0% (8 of 8 strings)

Translation: pos-16.0/pos-16.0-pos_session_pay_invoice
Translate-URL: https://translation.odoo-community.org/projects/pos-16-0/pos-16-0-pos_session_pay_invoice/it/

carlos-lopez-tecnativa committed:
[IMP] pos_session_pay_invoice: Restrict to journals of the current POS session
To prevent mistakes, users must select only the journals available in the pos.payment.method of the POS session.

Commit:
[UPD] Update pos_session_pay_invoice.pot

OCA-git-bot committed:
[BOT] post-merge updates

weblate committed:
Update translation files

Updated by "Update PO files to match POT (msgmerge)" hook in Weblate.

Translation: pos-16.0/pos-16.0-pos_session_pay_invoice
Translate-URL: https://translation.odoo-community.org/projects/pos-16-0/pos-16-0-pos_session_pay_invoice/

carlos-lopez-tecnativa committed:
[IMP] pos_session_pay_invoice: Generate pos.order and pos.payment for customer transactions

Before this commit, the wizard generated an account.bank.statement.line, but the problem was that when the POS had cash control enabled, all transactions generated by this wizard were displayed under Cash, because Odoo considered such statements as Cash In/Out operations, see https://github.com/odoo/odoo/blob/c1d3153ed564c0f47a201e5751740857a15d7db2/addons/point_of_sale/models/pos_session.py#L557-L567.

Now, the transactions are shown under Cash or Bank depending on the payment method and journal type.

For payments of vendor bills, a statement line will still be created, because the POS only works with receivable accounts and cannot reconcile with payable accounts.
See https://github.com/odoo/odoo/blob/c1d3153ed564c0f47a201e5751740857a15d7db2/addons/point_of_sale/models/pos_order.py#L890-L898

Commit:
[UPD] Update pos_session_pay_invoice.pot

OCA-git-bot committed:
[BOT] post-merge updates

weblate committed:
Update translation files

Updated by "Update PO files to match POT (msgmerge)" hook in Weblate.

Translation: pos-16.0/pos-16.0-pos_session_pay_invoice
Translate-URL: https://translation.odoo-community.org/projects/pos-16-0/pos-16-0-pos_session_pay_invoice/

mymage committed:
Translated using Weblate (Italian)

Currently translated at 100.0% (12 of 12 strings)

Translation: pos-16.0/pos-16.0-pos_session_pay_invoice
Translate-URL: https://translation.odoo-community.org/projects/pos-16-0/pos-16-0-pos_session_pay_invoice/it/

P-H-Phuc committed:
[IMP] pos_order_remove_line: Add config show button remove order line

P-H-Phuc committed:
[IMP] pos_payment_terminal: pre-commit auto fixes

P-H-Phuc committed:
[MIG] pos_payment_terminal: Migration to 18.0

nguyenminhchien reviewed (changes requested):
- Payment failed:
<img width="1866" height="968" alt="image" src="https://github.com/user-attachments/assets/b43cc90c-6e45-4a76-8197-6dac7088a408" />
- Add the unit tests

P-H-Phuc commented:
**SET UP FAKEBOX**
- Transaction started successfully:
   
<img width="809" height="236" alt="2025-12-22_18-02" src="https://github.com/user-attachments/assets/4bdfe119-dd05-422f-a280-fa4aafaafaca" />

- Transaction completed:
<img width="885" height="715" alt="2025-12-22_17-59" src="https://github.com/user-attachments/assets/a728dc43-51c7-4345-a2c9-575e1893e70c" />

**TEST CASE**

- Case 1: Test transaction amount is negative 
<img width="1396" height="656" alt="image" src="https://github.com/user-attachments/assets/516c5693-494e-4287-b832-83ef394f01ed" />

- Case 2: Test transaction start is false
<img width="1396" height="656" alt="image" src="https://github.com/user-attachments/assets/95432be3-1276-42ae-b0e2-f120c3540211" />

- Case 3: Test trasaction fail -> let the user retry
<img width="1396" height="656" alt="image" src="https://github.com/user-attachments/assets/4cbd37a3-9347-42b5-a6c8-b72004151a1d" />

nguyenminhchien reviewed (changes requested):
- Cannot test the module. it's always said: `Connecting to the payment terminal failed` when clicking on Payment method.
<img width="1859" height="965" alt="image" src="https://github.com/user-attachments/assets/c1a06843-f68f-44ec-87a3-d59690374fdd" />

nguyenminhchien commented on pos_payment_terminal/static/src/js/payment_terminal.esm.js:
```diff
@@ -0,0 +1,185 @@
+        if (this.payment_method_id.oca_payment_terminal_id) {
+            data.terminal_id = this.payment_method_id.oca_payment_terminal_id;
+        }
+        const response = await this._ocaPaymentTerminalProxyRequest(data);
```
Here's the function which handles the request in fakebox:
<img width="878" height="181" alt="image" src="https://github.com/user-attachments/assets/86f7d9c1-5024-439f-9a19-5a9f4f3e2a45" />

It returns `True`. So, it did not meet the condition `        if (!response) ` nor `else if (response instanceof Object && "transaction_id" in response)`.
So it always shows an error: `this._showError(_t("Connecting to the payment terminal failed"));`

@P-H-Phuc

nilshamerlinck replied:
MR welcome on fakebox codebase to support new module/odoo expectations ;-)

P-H-Phuc replied:
Hi @nilshamerlinck, @nguyenminhchien,
In v12.0 of Foodcoop and Fakebox, the endpoints for `pos_payment_terminal` currently simulate the [hw_telium_payment_terminal](https://github.com/AwesomeFoodCoops/posbox-addons/tree/master/hw_telium_payment_terminal) module. When calling `transaction_start`, no response is returned; the result is only available after calling `transaction_start_return`. And this response is an boolean.

In contrast, in later versions using `pos_payment_terminal` from` OCA/pos`, which rely on [pywebdriver](https://github.com/pywebdriver/pywebdriver) (it's `payment_mock_driver`), the result is returned immediately when `transaction_start` is called. Return a object which contains `transaction_id`

We are currently migrating to v18.0 on OCA. I have tested the functionalities, and they work correctly with [pywebdriver](https://github.com/pywebdriver/pywebdriver).

One key difference between the OCA version and Foodcoop is that Foodcoop uses custom, separate hardware modules, unlike OCA. This architectural difference makes testing more difficult.

nguyenminhchien replied:
@P-H-Phuc the module (12.0) is working fine on the live instances. you can make the changes during migration but you have to make sure it work well when connecting to the current terminal (as you mentioned, it currently simulates the [hw_telium_payment_terminal](https://github.com/AwesomeFoodCoops/posbox-addons/tree/master/hw_telium_payment_terminal) module)

Tisho99 committed:
[IMP] pos_session_pay_invoice: pre-commit auto fixes

Tisho99 committed:
[MIG] pos_session_pay_invoice: Migration to 18.0

P-H-Phuc commented:
**CONFIG**
<img width="1125" height="552" alt="image" src="https://github.com/user-attachments/assets/cd82081e-873f-4da3-a868-233b45fb7e8d" />
<img width="937" height="308" alt="image" src="https://github.com/user-attachments/assets/6404a902-725a-4f9c-8038-7026020ea7b3" />

@nguyenminhchien I made the changes to work with current terminal

nguyenminhchien commented:
@P-H-Phuc let me have a look. Please add the unit tests meanwhile.

nguyenminhchien commented on pos_payment_terminal/models/pos_payment_method.py:
```diff
@@ -0,0 +1,40 @@
+        string="Terminal Identifier",
+        help=("The identifier of the terminal as known by the hardware proxy. "),
+    )
+    hide_return_to_basket_btn_as_soon_as_payment_line_exists = fields.Boolean(
```
```suggestion
    oca_payment_terminal_hide_back_btn = fields.Boolean(
```

nguyenminhchien replied:
Add the usage of this field to the PR's description

nguyenminhchien commented on pos_payment_terminal/README.rst:
```diff
@@ -0,0 +1,124 @@
+
+In the frontend of the POS, when you select a payment method that has a
+payment mode with *Use a Payment Terminal* set to *OCA Payment
+Terminal*, you will have an additionnal *Send* button: if you click on
```
There's no button "Send".

P-H-Phuc replied:
When payment terminal mode is "Check", it will show "Send" button to wait user. fast_payments() is overrided to check that if payment mode is "Card" -> auto send, "Check" > show "Send" button

nguyenminhchien replied:
I didn't see this behavior in 16.0. Anw, if you want to stop sending payment automatically, you should better add a boolean field to indicate that behavior. It's explicit and easy to understand.

nguyenminhchien commented on pos_payment_terminal/static/src/js/models.js:
```diff
@@ -1,93 +0,0 @@
-                    if (
-                        payment_method.use_payment_terminal === "oca_payment_terminal"
-                    ) {
-                        this.config.use_proxy = true;
```
@P-H-Phuc in 16.0, it enabled the option to use the proxy here. but in 18.0, it trusts in `this.useProxy()`.
Please update the readme file to guide the reader how to enable the proxy.

P-H-Phuc replied:
I have some changes, I added a field iface_payment_terminal into Pos Config, if it is enabled > turn on Proxy in POS. And I also move "Hide Back Button" from PoS Payment Method to Pos Config. You can see it in USAGE.md

nguyenminhchien commented on pos_payment_terminal/static/src/app/payment_terminal.esm.js:
```diff
@@ -0,0 +1,180 @@
+import {register_payment_method} from "@point_of_sale/app/store/pos_store";
+
+export class OCAPaymentTerminal extends PaymentInterface {
+    get fast_payments() {
```
why need this? it returns `true` by default if using payment terminal, it's good enough.

nguyenminhchien replied:
@P-H-Phuc you may miss this comment!

nguyenminhchien commented on pos_payment_terminal/static/src/app/payment_terminal.esm.js:
```diff
@@ -0,0 +1,180 @@
+        return false;
+    }
+
+    send_payment_request(uuid) {
```
```suggestion
    async  send_payment_request(uuid) {
```

nguyenminhchien commented on pos_payment_terminal/static/src/app/payment_terminal.esm.js:
```diff
@@ -0,0 +1,180 @@
+    }
+
+    send_payment_request(uuid) {
+        super.send_payment_request(uuid);
```
```suggestion
        await super.send_payment_request(uuid);
```

nguyenminhchien replied:
There maybe more than 1 payment terminal, it should better wait for super.

nguyenminhchien commented on pos_payment_terminal/static/src/app/payment_terminal.esm.js:
```diff
@@ -0,0 +1,180 @@
+                this._ocaPollForTransactionStatus(line, resolve, reject);
+            });
+        }
+        return Promise.resolve(true);
```
```suggestion
        else if (response === true) return Promise.resolve(true);
        return Promise.resolve(false);
```

nguyenminhchien commented on pos_payment_terminal/static/src/app/payment_terminal.esm.js:
```diff
@@ -0,0 +1,180 @@
+            this.pos.hardwareProxy
+                .message("status_json", status_params)
+                .then((drivers) => {
+                    for (var driver_name in drivers) {
```
```suggestion
                    for (const driver_name in drivers) {
```

nguyenminhchien commented on pos_payment_terminal/static/src/app/payment_terminal.esm.js:
```diff
@@ -0,0 +1,180 @@
+                        if (!driver.is_terminal || !("transactions" in driver)) {
+                            continue;
+                        }
+                        for (var transaction_id in driver.transactions) {
```
```suggestion
                        for (const transaction_id in driver.transactions) {
```

nguyenminhchien commented on pos_payment_terminal/static/src/app/pos_store.esm.js:
```diff
@@ -0,0 +1,9 @@
+patch(PosStore.prototype, {
+    // @Override
+    useProxy() {
+        return this.config.iface_payment_terminal || super.useProxy();
```
take `is_posbox` into account.

nguyenminhchien commented on pos_payment_terminal/static/src/app/payment_terminal.esm.js:
```diff
@@ -0,0 +1,182 @@
+            paymentLine &&
+            paymentLine.payment_method_id.use_payment_terminal ===
+                "oca_payment_terminal" &&
+            paymentLine.payment_method_id.oca_payment_terminal_mode === "card"
```
I don't know why return True for `Card` but False for `Check`. I didn't see this behavior in 16.0.

P-H-Phuc replied:
This function is same with [get_automatic_payment_terminal](https://github.com/AwesomeFoodCoops/odoo-production/blob/12.0/pos_payment_terminal/static/src/js/models.js#L8-L15) in FoodCoop, I dont know clearly for this.

nguyenminhchien replied:
I see. If so, let add a new boolean field `Fast Payment`, default=True. User will untick if they need the "Send" button visible.

nguyenminhchien commented on pos_payment_terminal/static/src/app/payment_terminal.esm.js:
```diff
@@ -0,0 +1,182 @@
+        ) {
+            return true;
+        }
+        return false;
```
You even didn't handle the case of super, what happen if there's other payment terminal (stripe, xendit,..)?

nguyenminhchien commented on pos_payment_terminal/static/src/app/payment_terminal.esm.js:
```diff
@@ -0,0 +1,179 @@
+            paymentLine &&
+            paymentLine.payment_method_id.use_payment_terminal ===
+                "oca_payment_terminal" &&
+            paymentLine.payment_method_id.fast_payment
```
```suggestion
            paymentLine.payment_method_id.oca_fast_payment
```

nguyenminhchien replied:
Test it before pushing

nguyenminhchien reviewed (changes requested):
Add unit tests

Tisho99 committed:
[IMP] pos_margin: pre-commit auto fixes

Tisho99 committed:
[MIG] pos_margin: Migration to 18.0

BernatObrador committed:
[ADD] pos_user_restrict_provider_info: New module pos_user_restrict_provider_info

BernatObrador committed:
[ADD] pos_blind_session_closing: New module pos_blind_session_closing

OCA-git-bot committed:
Merge PR #1472 into 18.0

Signed-off-by pedrobaeza

Commit:
[UPD] Update pos_session_pay_invoice.pot

OCA-git-bot committed:
[BOT] post-merge updates

OCA-git-bot committed:
Merge PR #1296 into 18.0

Signed-off-by pedrobaeza

Commit:
[UPD] Update pos_order_remove_line.pot

OCA-git-bot committed:
[BOT] post-merge updates

OCA-git-bot committed:
Merge PR #1411 into 18.0

Signed-off-by pedrobaeza

Commit:
[UPD] Update pos_report_session_summary.pot

OCA-git-bot committed:
[BOT] post-merge updates

mymage committed:
Added translation using Weblate (Italian)

mymage committed:
Translated using Weblate (Italian)

Currently translated at 100.0% (11 of 11 strings)

Translation: pos-18.0/pos-18.0-pos_order_remove_line
Translate-URL: https://translation.odoo-community.org/projects/pos-18-0/pos-18-0-pos_order_remove_line/it/

OCA-git-bot committed:
Merge PR #1495 into 18.0

Signed-off-by ivantodorovich

Commit:
[UPD] Update pos_blind_session_closing.pot

OCA-git-bot committed:
[BOT] post-merge updates

OCA-git-bot committed:
Merge PR #1483 into 18.0

Signed-off-by ivantodorovich

Commit:
[UPD] Update pos_margin.pot

OCA-git-bot committed:
[BOT] post-merge updates

mymage committed:
Added translation using Weblate (Italian)

mymage committed:
Translated using Weblate (Italian)

Currently translated at 100.0% (1 of 1 strings)

Translation: pos-18.0/pos-18.0-pos_blind_session_closing
Translate-URL: https://translation.odoo-community.org/projects/pos-18-0/pos-18-0-pos_blind_session_closing/it/

OCA-git-bot committed:
Merge PR #1490 into 18.0

Signed-off-by ivantodorovich

Commit:
[UPD] Update pos_user_restrict_provider_info.pot

OCA-git-bot committed:
[BOT] post-merge updates

eduezerouali-tecnativa committed:
[MIG] pos_payment_method_cashdro: Migration to 18.0

OCA-git-bot committed:
Merge PR #1522 into 18.0

Signed-off-by pedrobaeza

Commit:
[UPD] Update pos_payment_method_cashdro.pot

OCA-git-bot committed:
[BOT] post-merge updates

eduezerouali-tecnativa committed:
[DOC] pos_payment_method_cashdro: Changes to roadmap

OCA-git-bot committed:
Merge PR #1524 into 18.0

Signed-off-by pedrobaeza

OCA-git-bot committed:
[BOT] post-merge updates

mymage committed:
Added translation using Weblate (Italian)

rousseldenis committed:
[UPD] Copier

[IMP] pos_order_to_sale_order: Added option to print sale pdf

[UPD] Update pos_order_to_sale_order.pot

[BOT] post-merge updates

Update translation files

Updated by "Update PO files to match POT (msgmerge)" hook in Weblate.

Translation: pos-18.0/pos-18.0-pos_order_to_sale_order
Translate-URL: https://translation.odoo-community.org/projects/pos-18-0/pos-18-0-pos_order_to_sale_order/

[FIX] pos_product_display_default_code: skip opening control in test tour

[BOT] post-merge updates

Initial check-in of the module pos_payment_terminal.

Add dependancy on currency_iso_numeric.

pos_customer_display bugfix : new massage on deleting orderLine and fixing unit_price computation on adding orderLine

Fix copyright headers
PEP8 stuff

pos_payment_terminal : bigger button for start transaction with touchscreen

pos_payment_terminal: add missing css file
pos_customer_display: FIX JS code and make it more robust

Remove currency_iso_numeric (replaced by pycountry)
Adapt JS code of pos_payment_terminal
Some cleanup

Add Spanish translation by Alejandro Amador (thanks for your contribution !)
Move CSS definition from pos_payment_terminal_view.xml to pos_payment_terminal.xml
Small cleanups

[FIX] Vicious bug that is invisible alone, but becomes visible when used with the module pos_pay_invoice

PEP8
